src/fastapi_app/dependencies.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import os
+from typing import Annotated
+
+import azure.identity
+from dotenv import load_dotenv
+from fastapi import Depends
+from openai import AsyncAzureOpenAI, AsyncOpenAI
+from pydantic import BaseModel
+from sqlalchemy.ext.asyncio import AsyncEngine, AsyncSession, async_sessionmaker
+
+from fastapi_app.openai_clients import create_openai_chat_client, create_openai_embed_client
+from fastapi_app.postgres_engine import create_postgres_engine_from_env
+
+logger = logging.getLogger("ragapp")
+
+
+class OpenAIClient(BaseModel):
+    """
+    OpenAI client
+    """
+
+    client: AsyncOpenAI | AsyncAzureOpenAI
+    model_config = {"arbitrary_types_allowed": True}
+
+
+class FastAPIAppContext(BaseModel):
+    """
+    Context for the FastAPI app
+    """
+
+    openai_chat_model: str
+    openai_embed_model: str
+    openai_embed_dimensions: int
+    openai_chat_deployment: str
+    openai_embed_deployment: str
+
+
+async def common_parameters():
+    """
+    Get the common parameters for the FastAPI app
+    """
+    load_dotenv(override=True)
```

Comment:
We now have load_dotenv in multiple places, is that strictly necessary? I thought we'd only need it in the lifespan function, which would then call the rest?

Reply:
It is necessary as this is now not managed by the application lifespan, if you call it in the application lifespan you don't have to do it, if you call it anywhere else say in the scripts for setting up the database you need to refresh the env vars as they are not gonna be loaded there.

For the application to work it's not necessary to call it for other uses of this function you need to call it.

Reply:
The only solution would be for me to go everywhere this function is being called and add/check that load_dotenv is  called before it.

Reply:
Also- The scripts do also call load_dotenv in their if == "main" logic, is that enough? Are there other situations where we call the functions outside of the lifespan/scripts?

Reply:
ok, I will review that and see if that's the case then it's not neededz

Reply:
you were correct, It's was not needed resolved here https://github.com/Azure-Samples/rag-postgres-openai-python/pull/55/commits/da5220c9431c9c8ec637157f86fc0f18d4fe989e
you were always loading env in the main and create_app.